subsys/bluetooth/mesh/sensor.c

```diff
@@ -623,7 +623,7 @@ const char *bt_mesh_sensor_ch_str(const struct sensor_value *ch)
 {
 	static char str[BT_MESH_SENSOR_CH_STR_LEN];
 
-	bt_mesh_sensor_ch_to_str(ch, str, sizeof(str));
+	__ASSERT_NO_MSG(bt_mesh_sensor_ch_to_str(ch, str, sizeof(str)) == 0);
```

Comment:
Are you sure it is correct? Aren't `bt_mesh_sensor_ch_to_str` which calls `snprintk` returns the number of bytes written? Also, if asserts are disabled, `bt_mesh_sensor_ch_to_str` will be removed: https://github.com/zephyrproject-rtos/zephyr/blob/a04a1d8991e312ece8530f0ab35b444010113fb9/include/zephyr/sys/__assert.h#L134-L143. We should never put function calls into assert macros.

Reply:
I have removed the assert for the case build with no asserts.